polyply/src/apply_links.py

```diff
@@ -453,11 +482,17 @@ def run_molecule(self, meta_molecule):
                     except MatchError as error:
                         LOGGER.debug(str(error), type='step')
 
+        # take care to remove nodes if there are any scheduled for removal
+        # we do this here becuase that's more efficent
+        molecule.remove_nodes_from(self.nodes_to_remove)
+        # now we add all interactions but not the ones that contain the removed
+        # nodes
         for inter_type in self.applied_links:
-            for interaction, citation in self.applied_links[inter_type].values():
-                meta_molecule.molecule.interactions[inter_type].append(interaction)
-                if citation:
-                    meta_molecule.molecule.citations.update(citation)
+            for atoms, (interaction, citation) in self.applied_links[inter_type].items():
+                if not any([atom in self.nodes_to_remove for atom in atoms]):
+                    meta_molecule.molecule.interactions[inter_type].append(interaction)
+                    if citation:
+                        meta_molecule.molecule.citations.update(citation)
```

Comment:
Is the if statement needed? Or can we always do `citations.update(citation)`? It'll depend on the default value of `citation`

Reply:
citation is an empty set so I think it should work